implementation/src/main/java/io/smallrye/mutiny/helpers/spies/MultiOnItemSpy.java

```diff
@@ -0,0 +1,48 @@
+package io.smallrye.mutiny.helpers.spies;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.subscription.MultiSubscriber;
+
+public class MultiOnItemSpy<T> extends MultiSpyBase<T> {
+
+    private final ArrayList<T> items = new ArrayList<>();
+
+    public List<T> items() {
+        List<T> view;
+        view = Collections.synchronizedList(items);
+        return view;
+    }
+
+    @Override
+    public void reset() {
+        super.reset();
+        synchronized (items) {
+            items.clear();
+        }
+    }
+
+    MultiOnItemSpy(Multi<? extends T> upstream) {
+        super(upstream);
+    }
+
+    @Override
+    public void subscribe(MultiSubscriber<? super T> downstream) {
+        upstream.onItem().invoke(item -> {
+            incrementInvocationCount();
+            synchronized (items) {
+                items.add(item);
```

Comment:
Should we add a parameter to not _store_ the item, to avoid OOM?

Reply:
I thought of that...